From d897d8176d13c964b5ffc9704b273f734cf56177 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Date: Fri, 31 Jul 2015 12:56:36 +0700
Subject: [PATCH 3/4] qtmultimedia: disable I420 support in qgstutils
 Qtrenderersink display wrong color for I420 format,
 remove I420 format from qgstutils and force
 Qtrenderersink use NV12

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
---
 qtmultimedia/src/gsttools/qgstutils.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gsttools/qgstutils.cpp b/src/gsttools/qgstutils.cpp
index 61ba09e..333bc2d 100644
--- a/src/gsttools/qgstutils.cpp
+++ b/src/gsttools/qgstutils.cpp
@@ -913,11 +913,11 @@ struct VideoFormat
 
 static const VideoFormat qt_videoFormatLookup[] =
 {
-    { QVideoFrame::Format_YUV420P, GST_VIDEO_FORMAT_I420 },
+    { QVideoFrame::Format_NV12   , GST_VIDEO_FORMAT_NV12 },
+    // { QVideoFrame::Format_YUV420P, GST_VIDEO_FORMAT_I420 },
     { QVideoFrame::Format_YV12   , GST_VIDEO_FORMAT_YV12 },
     { QVideoFrame::Format_UYVY   , GST_VIDEO_FORMAT_UYVY },
     { QVideoFrame::Format_YUYV   , GST_VIDEO_FORMAT_YUY2 },
-    { QVideoFrame::Format_NV12   , GST_VIDEO_FORMAT_NV12 },
     { QVideoFrame::Format_NV21   , GST_VIDEO_FORMAT_NV21 },
     { QVideoFrame::Format_AYUV444, GST_VIDEO_FORMAT_AYUV },
 #if Q_BYTE_ORDER == Q_LITTLE_ENDIAN
-- 
1.7.9.5

